dask/array/core.py

```diff
@@ -4269,13 +4269,11 @@ def stack(seq, axis=0, allow_unknown_chunksizes=False):
     if not seq:
         raise ValueError("Need array(s) to stack")
     if not allow_unknown_chunksizes and not all(x.shape == seq[0].shape for x in seq):
-        idx = np.where(np.asanyarray([x.shape for x in seq]) != seq[0].shape)[0]
+        idx = [x for x in filter(lambda x: x[1].shape != seq[0].shape, enumerate(seq))]
```

Comment:
- We try to avoid `filter` and filter in the comprehension for readability.
- By using `first` here (from `toolz`) we can avoid a full traversal of the arrays again.

This isn't equivalent (just returns the index into `seq`), requires change below as well.

```suggestion
        idx = first(i for i in enumerate(seq) if x.shape != seq[0].shape)
```

Reply:
Thanks for letting me know about `first`, that is a lot cleaner to use here.